### DISCUSSION TITLE
How to name PACTA_workflows?

### DISCUSSION THREAD

**Discussion by maurolepore** [2DegreesInvesting/PACTA_analysis#416] [General]
([Context](https://github.com/2DegreesInvesting/PACTA_analysis/pull/408#discussion_r586223619))

I propose to abandon the terms "online/offline workflow" because I find them misleading (e.g. you can run the "online" workflow offline). The alternatives I propose are "single-inputs" and "multiple-inputs" because the first one deals with a single investor and a single portfolio, whereas the second one deals with any number of investors and portfolios. But this new names may be not informative enough, and maybe the  whole "rename" thing is overly complicated -- I realize the multiple-inputs workflow should likely be the only workflow, with the single-inputs workflow being just a special case.

So, here are the questions:

1. Do you think we need a new name for "offline/online"?
2. Of so, which name do you propose?

**Comment by cjyetman** [comment]
I totally agree with moving away from naming these two processes "online" and "offline". They are inadequate and imprecise names.

In my opinion, if the goal is to name them something that is precise to what their intended purpose is (which I believe should be the goal), then the web tool scripts version should be called something like "web server" version, or "transition monitor" version. That's what its intended purpose is, and at this point sooooooo much depends on that being the case and remaining the case. I very much fear giving any other impression of it... anything that might convince someone to say something like "oh, maybe adjust this thing so it works better in single-portfolio mode" without full 100% recognition that this **has** to work for the server environment on transitionmonitor.com above all else.

so...
1. Yes
2. "transitionmonitor" and "PACTA"?

However, imo the even greater goal should be a convergence of these two processes into a single one so that there's no longer any confusion, nor need to maintain parallel versions, but that will require not only time and effort, but also an acknowledgement that getting the results of PACTA (merging the data and doing the calculations) and what one does with those results (do some analysis, create some plots, generate a PDF, create an interactive report) are two separate things, and the only reason we currently have two separate processes is because we think and talk about the web tool scripts as if its one full process, but in reality it is two separate processes than run in series. This is already reflected in how it's laid out: `web_tool_script_1.R` and `web_tool_script_2.R` do the PACTA process, and `web_tool_script_3.R` generates an executive summary and creates an interactive report. At some point we stopped thinking/talking about it like that though, and now we're a bit stuck because we consider that whole chain a unified process.
